services/iota_wrappers.go

```diff
@@ -458,35 +491,41 @@ func verifyChunksMatchRecord(chunks []models.DataMap, checkTrunkAndBranch bool)
 				continue
 			}
 			if _, ok := transactionObjects[chunkAddress]; ok {
-				matchFound := false
-				for _, txObject := range transactionObjects[chunkAddress] {
-					if chunksMatch(txObject, chunk, checkTrunkAndBranch) {
-						matchFound = true
-						break
-					}
-				}
-				if matchFound {
-					filteredChunks.MatchesTangle = append(filteredChunks.MatchesTangle, chunk)
-				} else {
-					filteredChunks.DoesNotMatchTangle = append(filteredChunks.DoesNotMatchTangle, chunk)
-				}
+
+				matchFound := checkTxObjectsForMatch(transactionObjects[chunkAddress], chunk, checkTrunkAndBranch)
+				assignToChunksArray(matchFound, chunk, &matchesTangle, &doesNotMatch)
 			} else {
-				filteredChunks.NotAttached = append(filteredChunks.NotAttached, chunk)
+				notAttached = append(notAttached, chunk)
 			}
 		}
-	} else if len(response.Hashes) == 0 {
-		filteredChunks.NotAttached = chunks
 	}
+	return matchesTangle, notAttached, doesNotMatch
+}
 
-	if len(filteredChunks.MatchesTangle) > 0 {
-		oyster_utils.LogToSegment("iota_wrappers: chunks_matched_tangle", analytics.NewProperties().
-			Set("num_chunks", len(filteredChunks.MatchesTangle)))
+func assignToChunksArray(matchFound bool, chunk models.DataMap, matchesTangle *[]models.DataMap, doesNotMatch *[]models.DataMap) {
```

Comment:
this method does not really add much value. Consider inline

Reply:
This was a codeclimate complaint.  I can change it back.